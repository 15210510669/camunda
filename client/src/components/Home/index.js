/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

export {default as Home} from './Home';
export {default as Collection} from './Collection';
