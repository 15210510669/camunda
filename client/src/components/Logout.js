/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {useEffect} from 'react';
import {withRouter} from 'react-router-dom';
import {get} from 'request';
import {withErrorHandling} from 'HOC';
import {addNotification} from 'notifications';
import {t} from 'translation';

export function Logout({mightFail, history}) {
  useEffect(() => {
    mightFail(
      get('api/authentication/logout'),
      () => {
        addNotification({text: t('navigation.logoutSuccess')});
        history.replace('/');
      },
      () => {
        addNotification({text: t('navigation.logoutFailed'), type: 'error'});
        history.replace('/');
      }
    );
  }, [mightFail, history]);

  return null;
}

export default withRouter(withErrorHandling(Logout));
