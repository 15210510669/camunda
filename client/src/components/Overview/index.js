export {default as Overview} from './Overview';
export {default as Alerts} from './Alerts';
