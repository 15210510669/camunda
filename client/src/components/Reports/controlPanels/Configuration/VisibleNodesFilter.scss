.VisibleNodesFilter {
  .Button {
    width: 100%;
  }
}
