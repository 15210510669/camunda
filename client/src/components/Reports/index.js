export {default as Reports} from './Reports';
export {default as Report} from './Report';
export {Filter} from './filter'
