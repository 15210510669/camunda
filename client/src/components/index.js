export {Footer} from './Footer';
export {Header, LogoutButton} from './Header';
export {Home} from './Home';
export {Login} from './Login';
export {PrivateRoute} from './PrivateRoute';
export {Dashboard} from './Dashboard';
export {Dashboards} from './Dashboards';
export {Report} from './Report';
export {Reports} from './Reports';
