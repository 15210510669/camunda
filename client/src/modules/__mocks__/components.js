export {default as Button} from '../components/__mocks__/Button';
export {default as Input} from '../components/__mocks__/Input';
export {default as Modal} from '../components/__mocks__/Modal';
export {default as Popover} from '../components/__mocks__/Popover';
export {default as ControlGroup} from '../components/__mocks__/ControlGroup';
export {default as ShareEntity} from '../components/__mocks__/ShareEntity';
export {default as DashboardView} from '../components/__mocks__/DashboardView';
export {default as Icon} from '../components/__mocks__/Icon';
export {default as Dropdown} from '../components/__mocks__/Dropdown';
export {default as Message} from '../components/__mocks__/Message';
export {default as ReportView} from '../components/__mocks__/ReportView';
export {default as Select} from '../components/__mocks__/Select';
export {default as BPMNDiagram} from '../components/__mocks__/BPMNDiagram';
export {default as TargetValueBadge} from '../components/__mocks__/TargetValueBadge';
export {default as Table} from '../components/__mocks__/Table';
export {default as Logo} from '../components/__mocks__/Logo';
export {default as ErrorMessage} from '../components/__mocks__/ErrorMessage';
