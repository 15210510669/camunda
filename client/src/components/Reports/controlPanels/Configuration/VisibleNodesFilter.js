/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';

import {Button, Switch} from 'components';
import NodeSelectionModal from './NodeSelectionModal';
import './VisibleNodesFilter.scss';
import {t} from 'translation';

export default class VisibleNodesFilter extends React.Component {
  state = {
    open: false
  };

  close = evt => this.setState({open: false});
  open = evt => this.setState({open: true});

  updateConfig = ({target: {checked}}) => {
    this.props.onChange({hiddenNodes: {active: {$set: checked}}});
  };

  render() {
    const {
      combined,
      data: {
        configuration: {hiddenNodes},
        groupBy
      }
    } = this.props.report;

    if (!combined && groupBy.type === 'flowNodes') {
      return (
        <div className="VisibleNodesFilter">
          <fieldset>
            <legend>
              <Switch
                label={t('report.config.visibleNodes.legend')}
                checked={hiddenNodes.active}
                onChange={this.updateConfig}
              />
            </legend>
            <Button disabled={!hiddenNodes.active} onClick={this.open}>
              {t('report.config.visibleNodes.btn')}
            </Button>
          </fieldset>
          {this.state.open && <NodeSelectionModal {...this.props} onClose={this.close} />}
        </div>
      );
    }
    return null;
  }
}
