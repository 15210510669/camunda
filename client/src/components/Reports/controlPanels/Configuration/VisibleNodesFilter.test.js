/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';

import VisibleNodesFilter from './VisibleNodesFilter';
import {shallow} from 'enzyme';
import {Button} from 'components';

const report = {
  result: {
    data: [{key: 'foo', value: 123}, {key: 'bar', value: 5}]
  },
  data: {
    configuration: {color: 'testColor', xml: 'fooXml', hiddenNodes: {active: false, keys: ['foo']}},
    visualization: 'line',
    groupBy: {
      type: 'flowNodes',
      value: ''
    },
    view: {}
  },
  targetValue: false,
  combined: false
};

it('should render nothing if report is not grouped by flowNodes', () => {
  const node = shallow(
    <VisibleNodesFilter
      report={{...report, data: {...report.data, groupBy: {type: 'something else'}}}}
    />
  );

  expect(node).toMatchSnapshot();
});
it('should render component', () => {
  const node = shallow(<VisibleNodesFilter report={report} />);

  expect(node).toMatchSnapshot();
});

it('should open NodeSelectionModal on show Flow Nodes button click', () => {
  const node = shallow(<VisibleNodesFilter report={report} />);

  node.find(Button).simulate('click');

  expect(node.find('NodeSelectionModal')).toExist();
});

it('Should invoke onChange when enabling the activation switch', () => {
  const spy = jest.fn();
  const node = shallow(<VisibleNodesFilter report={report} onChange={spy} />);

  expect(node.find(Button).props().disabled).toBe(true);

  node.find('Switch').simulate('change', {target: {checked: true}});

  expect(spy).toHaveBeenCalledWith({hiddenNodes: {active: {$set: true}}});
});
