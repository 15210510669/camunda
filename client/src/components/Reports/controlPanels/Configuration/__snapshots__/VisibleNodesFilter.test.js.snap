// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<div
  className="VisibleNodesFilter"
>
  <fieldset>
    <legend>
      <Switch
        checked={false}
        label="Customize Node Display"
        onChange={[Function]}
      />
    </legend>
    <ForwardRef(Button)
      disabled={true}
      onClick={[Function]}
    >
      Select Flow Nodes...
    </ForwardRef(Button)>
  </fieldset>
</div>
`;

exports[`should render nothing if report is not grouped by flowNodes 1`] = `""`;
